gstreamer-pbutils/src/auto/encoding_container_profile.rs

```diff
@@ -0,0 +1,68 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ fe7a6ff+)
+// from gir-files (https://github.com/gtk-rs/gir-files @ ???)
+// DO NOT EDIT
+
+use EncodingProfile;
+use ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct EncodingContainerProfile(Object<ffi::GstEncodingContainerProfile, ffi::GstEncodingContainerProfileClass>): EncodingProfile;
+
+    match fn {
+        get_type => || ffi::gst_encoding_container_profile_get_type(),
+    }
+}
+
+impl EncodingContainerProfile {
+    pub fn new<'a, 'b, 'c, P: Into<Option<&'a str>>, Q: Into<Option<&'b str>>, R: Into<Option<&'c str>>>(name: P, description: Q, format: &gst::Caps, preset: R) -> EncodingContainerProfile {
+        assert_initialized_main_thread!();
+        let name = name.into();
+        let name = name.to_glib_none();
+        let description = description.into();
+        let description = description.to_glib_none();
+        let preset = preset.into();
+        let preset = preset.to_glib_none();
+        unsafe {
+            from_glib_full(ffi::gst_encoding_container_profile_new(name.0, description.0, format.to_glib_none().0, preset.0))
+        }
+    }
+}
+
+unsafe impl Send for EncodingContainerProfile {}
+unsafe impl Sync for EncodingContainerProfile {}
+
+pub trait EncodingContainerProfileExt {
+    fn add_profile<P: IsA<EncodingProfile>>(&self, profile: &P) -> Result<(), glib::error::BoolError>;
```

Comment:
This one has API to modify it, but no mutexes seem to be used. So this can't be Sync, and because it's refcounted also not Send.

I would suggest to make all these objects immutable and move the mutable parts into a builder for them, so that they can be built up in the beginning but from that point onwards are immutable